support/util/networking_test.go

```diff
@@ -71,3 +71,85 @@ func TestGetAdvertiseAddress(t *testing.T) {
 		})
 	}
 }
+
+func TestIsDualStackHCP(t *testing.T) {
+	tests := []struct {
+		name           string
+		clusterNetwork *hyperv1.ClusterNetworking
+		want           bool
+	}{
+		{
+			name: "given an IPv4 SingleStack ServiceNetwork in the HCP, it should return false",
+			clusterNetwork: &hyperv1.ClusterNetworking{
+				ServiceNetwork: []hyperv1.ServiceNetworkEntry{{
+					CIDR: *ipnet.MustParseCIDR("192.168.1.0/24"),
+				}},
+			},
+			want: false,
+		},
+		{
+			name: "given an IPv6 SingleStack ServiceNetwork in the HCP, it should return false",
+			clusterNetwork: &hyperv1.ClusterNetworking{
+				ServiceNetwork: []hyperv1.ServiceNetworkEntry{{
+					CIDR: *ipnet.MustParseCIDR("2620:52:0:1306::1/64"),
+				}},
+			},
+			want: false,
+		},
+		{
+			name: "given an IPv4 SingleStack ClusterNetwork in the HCP, it should return false",
+			clusterNetwork: &hyperv1.ClusterNetworking{
+				ClusterNetwork: []hyperv1.ClusterNetworkEntry{{
+					CIDR: *ipnet.MustParseCIDR("192.168.1.0/24"),
+				}},
+			},
+			want: false,
+		},
+		{
+			name: "given an IPv6 SingleStack ClusterNetwork in the HCP, it should return false",
+			clusterNetwork: &hyperv1.ClusterNetworking{
+				ClusterNetwork: []hyperv1.ClusterNetworkEntry{{
+					CIDR: *ipnet.MustParseCIDR("2620:52:0:1306::1/64"),
+				}},
+			},
+			want: false,
+		},
+		{
+			name: "given an DualStack ServiceNetwork but not ClusterNetwork in the HCP, it should return false",
```

Comment:
could there be a scenario where a DualStack ClusterNetwork  is provided but not ServiceNetwork ? if so could we also test it.

Reply:
It could, we don't check that in our Network validations. I will submit a follow up PR with that validation. Good catch. In the meantime I will add the test to this PR. Thanks

Reply:
Having said that, I would say that OVN should fail setting the networking configuration.